### PR TITLE
Fixes score calculation

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/3134

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
defined in https://github.com/brave/brave-browser/issues/3134

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by NejcZdovc** [comment]
tried adding tests, but had problems testing private method with `friend` approach

File: vendor/bat-native-ledger/src/bat_publishers.cc [hunk 1]
```diff
@@ -262,7 +261,7 @@ void BatPublishers::saveVisitInternal(
              verified_old) {
     publisher_info->visits += 1;
     publisher_info->duration += duration;
-    publisher_info->score += concaveScore(duration);
+    publisher_info->score += concaveScore(duration * 1000);
```
Comment (bridiver) [inline vendor/bat-native-ledger/src/bat_publishers.cc]: why is this `* 1000` we should specify the units `duration_in_seconds` or `duration_in_milliseconds` or whatever it is

File: vendor/bat-native-ledger/src/bat_publishers.cc [hunk 2]
```diff
@@ -35,25 +35,24 @@ BatPublishers::BatPublishers(bat_ledger::LedgerImpl* ledger):
...
 BatPublishers::~BatPublishers() {
 }
 
-void BatPublishers::calcScoreConsts() {
-  uint64_t min_duration_ms = state_->min_publisher_duration_ * 1000;
-  //TODO: check Warning	C4244	'=': conversion from 'double' to 'unsigned int', possible loss of data
-  a_ = 1.0 / (braveledger_ledger::_d * 2.0) - min_duration_ms;
-  a2_ = a_ * 2;
-  a4_ = a2_ * 2;
+void BatPublishers::calcScoreConsts(const uint64_t& min_duration) {
```
Comment (bridiver) [inline vendor/bat-native-ledger/src/bat_publishers.cc]: side note unrelated to this PR: `calcScoreConsts` doesn't seem like a great name. If we are updating the values they are clearly not constants

File: vendor/bat-native-ledger/src/bat_publishers.cc [hunk 3]
```diff
@@ -529,6 +528,11 @@ void BatPublishers::synopsisNormalizerInternal(
   }
   double totalScores = 0.0;
   for (size_t i = 0; i < list.size(); i++) {
+    // Check which would test uint problem from this issue
+    // https://github.com/brave/brave-browser/issues/3134
+    if (list[i].score < -100000) {
```
Comment (bridiver) [inline vendor/bat-native-ledger/src/bat_publishers.cc]: why `-100000`?

File: vendor/bat-native-ledger/src/bat_publishers.cc (hunk 1 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat_publishers.cc]: done

**Pull request merged**